crucible/src/Lang/Crucible/Simulator/Breakpoint.hs

```diff
@@ -0,0 +1,74 @@
+-----------------------------------------------------------------------
+-- |
+-- Module           : Lang.Crucible.Simulator.Breakpoint
+-- Description      : Support for symbolic execution breakpoints
+-- Copyright        : (c) Galois, Inc 2019
+-- License          : BSD3
+-- Maintainer       : Andrei Stefanescu <andrei@galois.com>
+-- Stability        : provisional
+--
+-- This module provides an execution feature for bounding the
+-- number of iterations that a loop will exeucte in the simulator.
+-----------------------------------------------------------------------
+{-# LANGUAGE GADTs #-}
+{-# LANGUAGE LambdaCase #-}
+{-# LANGUAGE RecordWildCards #-}
+module Lang.Crucible.Simulator.Breakpoint
+  ( breakAndReturn
+  ) where
+
+import           Control.Lens
+import           Control.Monad.Reader
+import qualified Data.Bimap as Bimap
+
+import           Data.Parameterized.Classes
+import qualified Data.Parameterized.Context as Ctx
+import           Data.Parameterized.Some
+import           Data.Parameterized.TraversableFC
+
+import qualified Lang.Crucible.Backend as C
+import qualified Lang.Crucible.CFG.Core as C
+import qualified Lang.Crucible.CFG.Expr as C
+import qualified Lang.Crucible.Simulator.CallFrame as C
+import qualified Lang.Crucible.Simulator.EvalStmt as C
+import qualified Lang.Crucible.Simulator.ExecutionTree as C
+import qualified Lang.Crucible.Simulator.Operations as C
+import qualified Lang.Crucible.Simulator.OverrideSim as C
+import qualified Lang.Crucible.Simulator.RegValue as C
+import qualified What4.FunctionName as W
+
+breakAndReturn
```

Comment:
In general, I'm guessing you actually need a mapping from breakpoint names to overrides?

Reply:
Currently, each execution feature handles one breakpoint. This is ok if there are only a few breakpoints. I can add a map later.